firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/messages.txt

```diff
@@ -0,0 +1,181 @@
+/* Write function
+Scenario: Message: The write method correctly generates an encoded body from the protobuf message descriptor payload
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes")
+  // assertions carried out in given:
+  // output buffer is empty
+
+    WHEN("The value of the message payload tag is greater than the size of the message descriptor array")
+      THEN("The write method reports invalid type status")
+      THEN("After the write method is called, the output buffer remains unchanged")
+
+    WHEN("The message payload tag is such that it corresponds to Unrecognized message descriptor in descriptor array")
+      THEN("The write method reports invalid type status")
+      THEN("After the write method is called, the output buffer remains unchanged")
+
+    WHEN("The output buffer is too small to hold the generated body")
+      THEN("The write method reports invalid length status")
+      THEN("After the write method is called, the output buffer remains unchanged")
+
+    WHEN("The size of the output buffer is zero")
+      THEN("The write method reports invalid length status")
+
+    // Intent: To show that the write method does not change the internal "type" member of the message class
+    WHEN("The internal type member is set to an arbitrary value and then write method is called")
+    // setup:
+    // type is set to 4
+    // A sensor measurements payload message is set, which has value of type as 2
+      THEN("The write method reports ok status")
+      THEN("After the write method is called, type member remains unchanged)
+
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 126 bytes")
+    WHEN("The payload is a sensor measurements message and write method is called")
+      THEN("The write method reports ok status")
+      THEN("The body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer is as expected")
```

Comment:
I mentioned in the review of #297 that it could be good to describe the output buffer (e.g. by giving its expected length), and I think it would be useful here too